dist/profile/templates/buildmaster/casc/clouds.yaml.erb

```
@@ -160,7 +160,7 @@ jenkins:
         - containers:
           - alwaysPullImage: false
             image: "<%= @agent_images["container_images"][agent["name"]] %>"
-            env:
+            envVars:
             - name: "JENKINS_JAVA_OPTS"
               value: "-XX:+PrintCommandLineFlags"
```

```suggestion
            envVars:
              - envVar:
                   key: "JENKINS_JAVA_OPTS"
                   value: "-XX:+PrintCommandLineFlags"
```

As per https://github.com/jenkinsci/configuration-as-code-plugin/issues/204#issuecomment-389623788 

(I did the same as you and had another stacktrace. Tried with the local vagrant VM used to test the puppet template, went through the Jenkins UI, set up manually the env vars, and checked the jcasc export).